profiles.wordpress.org/public_html/wp-content/plugins/wporg-5ftf-activity-handler/wporg-5ftf-activity-handler.php

```diff
@@ -0,0 +1,136 @@
+<?php
+/*
+Plugin Name: WordPress.org 5ftf Activity Handler
+Plugin URI: http://wordpress.org
+License: GPL2
+Version: 1.1
+Description: Handles saving of last 5ftf contribution.
+*/
+
+defined( 'ABSPATH' ) or die();
+
+if ( class_exists( 'WPOrg_5ftf_Activity_Handler' ) ) {
+	return;
+}
+
+class WPOrg_5ftf_Activity_Handler {
+
+	/**
+	 * @var WPOrg_5ftf_Activity_Handler The singleton instance.
+	 */
+	private static $instance;
+
+	/**
+	 *  Name of meta key that tracks last contribution as a unix timestamp.
+	 *
+	 * @var string
+	 */
+	const last_contribution_meta_key = 'wporg_5ftf_last_contribution';
+
+	/**
+	 *  Name of meta key that stores information about the last_contribution_meta_key.
+	 *
+	 * @var string
+	 */
+	const last_contribution_info_meta_key = 'last_contribution_info_meta_key';
+
+	/**
+	 * Returns always the same instance of this plugin.
+	 *
+	 * @return WPOrg_5ftf_Activity_Handler
+	 */
+	public static function get_instance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		if ( 'profiles.wordpress.org' === site_url() ) {
+			add_filter( 'bp_activity_add', array( $this, 'handle_activity' ) );
+		} elseif ( str_ends_with( 'wordpress.org', site_url() ) ) {
```

Comment:
Oh, before merging it'd be best to [add the `IS_WPORG_MULTINETWORK` bit](https://github.com/WordPress/wordpress.org/pull/78#discussion_r908806352), though, or at least a comment explaining that this matches all of the sites in the w.org multinetwork (*.w.org). It's kind of confusing because `profiles.wordpress.org` is actually on the _buddypress_ network, even though it has a w.org hostname.

WIthout some kind of explanation the "if .org else .org" is confusing at a quick glance.

Reply:
Oh right, sorry I missed that. Added a comment back in the original thread.

https://github.com/WordPress/wordpress.org/pull/78#discussion_r908806352